homework-submission/myapp4/src/stores/TodosStore.js

```diff
@@ -0,0 +1,74 @@
+import { observable, computed, action, runInAction } from "mobx";
+
+const Parent_API = "https://hyf-react-api.herokuapp.com";
+
+
+class TodosStore {
+  @observable Todos = {
+    data: [],
+    status: "loading..."
+  };
+
+  @action createTodo(todo) {
+    this.createNewTodo.status = "loading...";
+    this.createNewTodo(todo)
+      .then(todo => {
+        runInAction(() => {
+          this.Todos.data.push(todo);
+          this.Todos.status = "done";
+        });
+      })
+      .catch(error => {
+        runInAction(() => {
+          this.Todos.status = "error";
+        });
+      });
+  }
+
+  @action async addTodos() {
+    this.Todos.data = [];
+    this.addTodos.status = "loading...";
+
+    try {
+      const addedTodo = await this.getTodos();
+      runInAction(() => {
+        this.Todos.data = addedTodo;
+        this.Todos.status = "done";
+      });
+    } catch (error) {
+      runInAction(() => {
+        this.Todos.status = "error";
+      });
+    }
+  }
+    
+  @action removeTodo = (id) => {
+      this.Todos.data.splice(id, 1);
+  };
+  
+  @action isDone = todo => {
+    todo.done = !todo.done;
```

Comment:
This is giving a client side behavior without an actual Effect from the server it self